packages/website/public/llms/styleGuide.md

```diff
@@ -0,0 +1,417 @@
+  Examples:
+
+```squiggle
+@name("Number of Humans (2023)")
+number_of_humans = 7.8B
+
+@name("Net Benefit ($)")
+net_benefit = 100M
+
+@name("Temperature (°C)")
+temperature = 22
+
+@name("Piano Tuners in New York City (2023)")
+tuners = {
+    pianos_per_piano_tuners = 100 to 1k // (pianos per tuner)
+    pianos_in_nyc = 1k to 50k // (pianos)
+    pianos_in_nyc / pianos_per_piano_tuners
+}
+```
+
+- Maintain Consistent Units. Ensure that related variables use the same units to prevent confusion and errors in calculations.
+
+```squiggle
+@name("Distance to Mars (km)")
+distance_mars = 225e6
+
+@name("Distance to Venus (km)")
+distance_venus = 170e6
+```
+
+### Numbers
+
+- Use abbreviations, when simple, for numbers outside the range of 10^4 to 10^3. For example, use "10k" instead of "10000".
+- For numbers outside the range of 10^10 or so, use scientific notation. For example, "1e10".
+- Don't use small numbers to represent large numbers. For example, don't use '5' to represent 5 million.
+
+Don't use the code:
+
+```squiggle
+@name("US Population (millions)")
+us_population = 331.9
+```
+
+Instead, use:
+
+```squiggle
+@name("US Population")
+us_population = 331.9M
+```
+
+More examples:
+
+```squiggle
+// Correct representations
+world_population = 7.8B
+annual_budget = 1.2T
+distance_to_sun = 149.6e6  // 149.6 million kilometers
+
+// Incorrect representations (avoid these)
+world_population = 7800  // Unclear if it's 7800 or 7.8 billion
+annual_budget = 1200  // Unclear if it's 1200 or 1.2 trillion
+```
+
+- There's no need to use @format on regular numbers. The default formatting is fairly sophistated.
+
+### Lists of Structured Data
+
+- When you want to store complex data as code, use lists of dictionaries, instead of using lists of lists. This makes things clearer. For example, use:
+
+```squiggle
+[
+  {year: 2023, value: 1},
+  {year: 2024, value: 2},
+]
+instead of:
+[
+  [2023, 1],
+  [2024, 2],
+]
+```
+
+You can use lists instead when you have a very long list of items (20+), very few keys, and/or are generating data using functions.
+
+- Tables are a great way to display structured data.
+- You can use the '@showAs' tag to display a table if the table can show all the data. If this takes a lot of formatting work, you can move that to a helper function. Note that helper functions must be placed before the '@showAs' tag.
+
+For example:
+
+```squiggle
+@hide
+strategiesTable(data) = Table.make(
+  data,
+  {
+    columns: [
+      { name: "name", fn: {|f| f.n} },
+      { name: "costs", fn: {|f| f.c} },
+      { name: "benefits", fn: {|f| f.b} },
+    ],
+  }
+)
+
+@name("AI Safety Strategies")
+@doc("List of 10 AI safety strategies with their costs and benefits")
+@showAs(strategiesTable)
+strategies = [
+  { n: "AI Ethics", c: 1M to 5M, b: 5M to 20M },
+  { n: "Alignment Research", c: 2M to 10M, b: 10M to 50M },
+  { n: "Governance", c: 500k to 3M, b: 2M to 15M },
+  ...
+]
+```
```

Comment:
This made me curious if I can automate it, so, here: https://squigglehub.org/models/berekuk/autoTable

Reply:
Thanks! I added that to the helpers library.